src/analysis/normed_space/real_inner_product.lean

```diff
@@ -0,0 +1,174 @@
+/-
+	Copyright (c) 2019 Zhouhang Zhou. All rights reserved.
+	Released under Apache 2.0 license as described in the file LICENSE.
+	Authors: Zhouhang Zhou
+  -/
+
+import analysis.normed_space.basic
+import analysis.normed_space.discriminant
+
+/-!
+	# Inner Product Space
+
+  Define inner product space over reals and prove its basic properties.
+
+	## Implementation notes
+
+	## Tags
+
+	inner product space, norm
+
+	## References
+  *  [Clément & Martin, *The Lax-Milgram Theorem. A detailed proof to be formalized in Coq*]
+	*  [Clément & Martin, *A Coq formal proof of the Lax–Milgram theorem*]
+
+	The Coq code is available at the following address: http://www.lri.fr/~sboldo/elfic/index.html
+  -/
+
+noncomputable theory
+
+universes u v w
+
+variables {α : Type u} {F : Type v} {G : Type w}
+
+class has_inner (α : Type*) := (inner : α → α → ℝ)
```

Comment:
@amswerdlow used the notation `ₕ⟨x|y⟩` to mean the inner product of `x` and `y`, but I feel `inner x y` is better.

Reply:
I agree -- `inner x y` is easier to remember and type. I wonder how others feel about this.